File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 6/2/2025, 6:43:26 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> m;
        for (int i = 0;; ++i) {
            int x = nums[i];
            int y = target - x;
            if (m.count(y)) {
                return {m[y], i};
            }
            m[x] = i;
        }
    }
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 7/19/2025, 10:16:35 AM
from typing import List
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n= len(nums)
        for i in range(n):
            for j in range(i+1,n):
                if nums[i]+nums[j] == target:
                    return [i,j]
        return[]
obj=Solution()
print(obj.twoSum([2,7,11,15],9))            